cpp/arduino/Arduino.h

```diff
@@ -5,6 +5,15 @@ Mock Arduino.h library.
 Where possible, variable names from the Arduino library are used to avoid conflicts
 
 */
+
+// hardware mocks
+volatile long long __ARDUINO_CI_SFR_MOCK[1024];
+#define _SFR_IO8(io_addr) (*(volatile uint8_t *)(__ARDUINO_CI_SFR_MOCK + io_addr)) // this macro is all we need from the sfr file
+#define _SFR_IO16(io_addr) (*(volatile uint16_t *)(__ARDUINO_CI_SFR_MOCK + io_addr)) // this macro is all we need from the sfr file
+#define _SFR_MEM8(io_addr) (*(volatile uint8_t *)(__ARDUINO_CI_SFR_MOCK + io_addr)) // this macro is all we need from the sfr file
+#define _SFR_MEM16(io_addr) (*(volatile uint16_t *)(__ARDUINO_CI_SFR_MOCK + io_addr)) // this macro is all we need from the sfr file
+#define _SFR_MEM32(io_addr) (*(volatile uint32_t *)(__ARDUINO_CI_SFR_MOCK + io_addr)) // this macro is all we need from the sfr file
```

Comment:
I would leave this stuff in `io.h`, which is where they are used and what any other code that (god knows why) uses them would expect them.

Reply:
I'm looking at this and right now the issue is that `SampleProjects/TestSomething/test/godmode.cpp` fails the `spi` tests due to bytes being in the wrong order.

Reply:
I'm not sure what you mean here? If you move this to io.h, that makes the tests fail? But not because of include order or something like that, but "byte order"?

Reply:
Just for completeness, this conversation is continuing in #201.